tests/test_location.py

```diff
@@ -181,6 +181,22 @@ def testButlerUriSerialization(self):
         self.assertEqual(uri, uri2)
         self.assertTrue(uri2.dirLike)
 
+    def testUriExtensions(self):
+        """Test extension extraction."""
+
+        files = (("file.fits.gz", ".fits.gz"),
+                 ("file.fits", ".fits"),
+                 ("file.fits.xz", ".fits.xz"),
+                 ("file.fits.tar", ".tar"),
+                 ("file", ""),
+                 ("flat_i_sim_1.4_blah.fits.gz", ".fits.gz"),
+                 ("flat_i_sim_1.4_blah.txt", ".txt"),
+                 )
```

Comment:
If we do support `.fits` and `.fits.fz` then I think that we should explicitly test that here.

Reply:
Since `.fz` is not listed as a special extension at the moment getExtension() will return `.fz` for `.fits.fz`.

Reply:
I think that's fine!  I would suggest putting those in the list so that it's clear what it will return (and that we know it's returning what we think it should return).